src/styles/main.scss

```diff
@@ -0,0 +1,61 @@
+$day-size: 100px;
+$gap-size: 1px;
+$columns: 7;
+$padding-size: 10px;
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: $padding-size;
+  gap: $gap-size;
+  max-width: calc(
+    #{$columns} * #{$day-size} + (#{$columns} - 1) * #{$gap-size} + 2 * #{$padding-size}
+  );
+
+  .day {
+    display: flex;
+    justify-content: center;
+    box-sizing: border-box;
+    height: $day-size;
+    width: $day-size;
+    background-color: #eee;
+    border: 1px solid black;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    transition-duration: 500ms;
+
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-of-type(#{$i})::before {
+        content: '#{$i}';
+        display: flex;
+        justify-content: center;
+        align-items: center;
+      }
+    }
+  }
+
+  @each $day,
+    $index
+      in ('mon': 1, 'tue': 2, 'wed': 3, 'thu': 4, 'fri': 5, 'sat': 6, 'sun': 7)
+  {
+    &.calendar--start-day-#{$day} {
+      .day {
+        &:nth-child(1) {
+          margin-left: calc((#{$index} - 1) * (#{$day-size} + #{$gap-size}));
```

Comment:
Avoid using `calc` with SASS variables if the calculation can be done directly in SASS, as recommended in the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules).